src/lib/dnssd/minimal_mdns/responders/IP.cpp

```diff
@@ -16,18 +16,24 @@
  */
 #include "IP.h"
 
+#include <lib/dnssd/minimal_mdns/AddressPolicy.h>
 #include <lib/dnssd/minimal_mdns/records/IP.h>
 
 namespace mdns {
 namespace Minimal {
 
+using chip::Platform::UniquePtr;
+
 void IPv4Responder::AddAllResponses(const chip::Inet::IPPacketInfo * source, ResponderDelegate * delegate,
                                     const ResponseConfiguration & configuration)
 {
     chip::Inet::IPAddress addr;
-    for (chip::Inet::InterfaceAddressIterator it; it.HasCurrent(); it.Next())
+    UniquePtr<IpAddressIterator> ips = Policies::GetIpAddressesForEndpoint(source->Interface);
+    VerifyOrDie(ips);
```

Comment:
Please just early exit on failure to alloc. Crashing is quite harsh

Reply:
Returning no IP addresses seems very hard to debug. Are you sure it is a better way of working?

Reply:
So thinking of options:
  - apply `ostrich approach/head in sand` and just drop it. If we run out of mem (not as likely as these objects are basically a vtable) then we will crash on vtable access Next(). I think we do that in other places.
  - This explicit crash as we have now (ugly to crash, but explicit crash location at least).
  - if condition with nice log of 'memory allocation error'. Seemed overkill.